packages/block-library/src/buttons/style.scss

```diff
@@ -18,10 +29,16 @@
 
 	&.is-content-justification-left {
```

Comment:
Perhaps this className should have been `is-content-justification-start`, because "left" doesn't make sense in the vertical context, and the actual value of `align-items` that gets set is `flex-start` anyway. Same for `is-content-justification-right` which should probably be `is-content-justification-end`.

Reply:
Just realized the above comment was incorrect, since I got horizontal/vertical controls confused in my head.

Anyway, there is an actual problem here, though: this class is being used to `align-items` for vertical Buttons blocks, but the class name implies it can only set `justify-content`. Additionally, the `justify-content` styles are being applied to both vertical and horizontal orientations. That's not technically correct.

Reply:
Why not? You can have vertically stacked buttons and align them for example to the right. What do you propose exactly?

Reply:
We shouldn't be setting `justify-content` for the vertical case... only `align-items`. But also, the class "is-content-justification-xxx" is kind of confusing if it's actually applying item alignment in the vertical cases. I think ideally, we'd want to tackle this stuff sooner rather than later to reduce the number of deprecations and legacy styles we have to keep on the front-end.

Reply:
The problem here is basically that I named the `is-content-justification-xxx` classes with the assumption that they would only be used to apply different `justify-content` styles, but that's no longer true in this case.

Maybe a new class name that can work both cases would be appropriate. I'm not sure what the correct term is, though.

Or maybe the solution is to apply different classes for the vertical cases, e.g. `is-item-alignment-xxx`. I'm not sure.

Reply:
This is gnarly, and it seems like there are several things going on:

- The desire for a more abstracted system of defining flex containers (CC: @youknowriad as I know you've thought about this)
- The fact that in flex, the two properties `align-items` and `justify-content` center vertically and horizontally, depending on which `flex-direction` is applied.

While both of those are valid, I think maybe we're establishing properties here that don't have anything to do with the underlying tech, which could be flex, or CSS grid, or something else. In that vein, personally I think of _alignment_ as the positioning of the block itself, in this case the container block, and _justificiation_ as the positioning of content inside. 

Is that the path forward? Deciding on some classnames that have nothing to do with the CSS that is used, but only with the editor visual semantics?

Reply:
Agreed here, I don't think we should be naming classnames based on technology but more based on intent and yes it seems we want to "align content" with this. It's called now "justify content" which happens to map with "justify-content" on the default flex orientation. I don't really care too much that it doesn't match the naming of the  CSS property in vertical orientation and I don't really think we should be renaming the classname just for that.

Reply:
> personally I think of _alignment_ as the positioning of the block itself, in this case the container block, and _justificiation_ as the positioning of content inside.

It's worth remembering that in addition to `align-items` and `justify-content`, there's also [`align-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/align-content) and [`justify-items`](https://developer.mozilla.org/en-US/docs/Web/CSS/justify-items), though `justify-items` doesn't do anything in Flexbox.

By looking at how these properties work, we can learn the following about how CSS is using these terms:

In a horizontally-flowing container, "align" refers to vertical positioning and "justify" refers to horizontal positioning. In a vertically-flowing container, "align" refers to horizontal positioning and "justify" refers to vertical positioning. In other words, "justify" refers to positioning along the primary axis that content is flowing, and "align" refers to the secondary (AKA "cross") axis.

"content" applies to all the children as a single block within the container, and "items" applies to each child as an individual unit.

Technically, the kind of alignment/justification we want for the horizontal variant is actually conceptually different from the control we want on the vertical variant. Technically, `justify-content`'s cross-axis complement is `align-content`, not `align-items`. :confounded:

Reply:
If I had to come up with a better class name than what we're using right now, I guess something like `is-content-horizontal-placement-xxx` might work. I'm drawing the "placement" term here from the shorthand properties [`place-content`](https://developer.mozilla.org/en-US/docs/Web/CSS/place-content) and [`place-items`](https://developer.mozilla.org/en-US/docs/Web/CSS/place-items). The CSS "placement" term is apparently its generic way of referring to both "alignment" and "justification".

Of course, in my suggested class name here, the meaning of "content" is slightly different from what CSS uses it for, since in the case of my suggested class name, it can refer to either CSS's "content" or "items". But I guess it's "good enough" for our purposes?

Reply:
For me it's too late to change the class names, this is already shipped and themes might be relying on it. The renaming doesn't seem to add a lot of value.

Reply:
As far as I can tell, it's only shipped in the plugin, not core, right? Does that change anything?

Reply:
> Does that change anything?

It does, we have more luxury to change things that way but are we certain of that? It's also a bit late if it's in WP 5.6 as we're on RC2 and the stable is supposed to happen next week. 

We need to do a few things if we want to rename the classnames:

 - try to check usage of the classname in the plugins/theme repository
 - backport the fix to  WP 5.6 (so it needs to be in its own separate PR and not block this one)

cc @tellthemachines